Remove 75% of explicit scalings by 1/128 in the AArch64 inverse NTT

hanno-becker opened the pull request:
<!-- 
Security reports

DO NOT submit pull requests related to security issues directly - instead use Github's [private vulnerability reporting](https://github.com/pq-code-package/mlkem-native/security). 
-->

**Summary**:
Provide a brief summary of your changes.

**Steps**:
If your pull request consists of multiple sequential changes, please describe them here:

**Performed local tests**:
 - [ ] `lint` passing
 - [ ] `tests all` passing
 - [ ] `tests bench` passing
 - [ ] `tests cbmc` passing

**Do you expect this change to impact performance**: Yes/No

If yes, please provide local benchmarking results.

hanno-becker committed:
Remove 75% of explicit scalings by 1/128 in the AArch64 inverse NTT

The AArch64 inverse NTT includes an explicit by 1/128 at its
beginning. Following 6.4.5 of

  Polynomial Multiplication on Embedded Vector Architectures
  https://eprint.iacr.org/2021/998

we integrate 75% of those scalings into the twiddle factors
in the last two layers.

For the last layer, this requires an additional twiddle, which
can conveniently be stored in memory previously used for padding.

In theory, one could save more scalings, but the logistics of this
get harder and require more twiddles -- so it's not attempted.

The HOL-Light proof adjusted.

Signed-off-by: Hanno Becker <beckphan@amazon.co.uk>

mkannwischer reviewed (changes requested):
Valgrind is failing with 
```
==5615== Copyright (C) 2002-2024, and GNU GPL'd, by Julian Seward et al.
  ==5615== Using Valgrind-3.23.0 and LibVEX; rerun with -h for copyright info
  ==5615== Command: test/build/mlkem512/bin/test_mlkem512
  ==5615== 
  
  vex: priv/host_arm64_defs.c:2831 (genSpill_ARM64): Assertion `offsetB < 4096' failed.
  vex storage: T total 2148258[72](https://github.com/pq-code-package/mlkem-native/actions/runs/14012016795/job/39233102170?pr=903#step:4:78) bytes allocated
  vex storage: P total 0 bytes allocated
  
  valgrind: the 'impossible' happened:
     LibVEX called failure_exit().
```

HOL-Light fails with 
```
    Fatal error: exception Failure("dest_cons4: 4-byte inst code 1384136709 != first 4 bytes of [word 227; word 3; word 0; word 170; word 4; word 1; word 128; word 210;\n word 107; word 0; word 192; word 61; word 106; word 4; word 192; word 61;\n word 113; word 12; word 192; word 61; word 117; word 8; word 192; word 61;\n word 99; word 105; word 138; word 78; word 106; word 41; word 138; word 78;\n word 187; word 42; word 145; word 78; word 77; word 20; word 192; word 61;\n word 70; word 105; word 219; word 78; word 169; word 106; word 145; \n word 78; word 88; word 41; word 219; word 78; word 85; word 8; word 192;\n word 61; word 116; word 40; word 201; word 78; word 108; word 104; word 201;\n word 78; word 1; word 135; word 116; word 110; word 195; word 132; word 108;\n word 110; word 91; word 12; word 192; word 61; word 94; word 16; word 192;\n word 61; word 42; word 180; word 123; word 110; word 109; word 180;\n word 109; word 110; word 110; word 156; word 126; word 78; word 43;\n word 156; word 117; word 78; word 213; word 132; word 108; word 78; \n word 1; word 135; word 116; word 78; word 174; word 65; word 71; word 111;\n word 75; word 65; word 71; word 111; word 74; word 4; word 198; word 60;\n word 52; word 132; word 117; word 110; word 125; word 133; word 110;\n word 110; word 89; word 0; word 219; word 60; word 187; word 159; word 106;\n word 78; word 144; word 158; word 106; word 78; word 173; word 183;\n word 121; word 110; word 138; word 182; word 121; word 110; word 37;\n word 132; word 117; word 78; word 127; word 133; word 110; word 78;\n word 187; word 65; word 71; word 111; word 80; word 65; word 71; word 111;\n word 43; word 4; word 193; word 60; word 181; word 104; word 159; word 78;\n word 10; word 106; word 155; word 78; word 6; word 42; word 155; word 78;\n word 165; word 40; word 159; word 78; word 173; word 42; word 202; word 78;\n word 181; word 106; word 202; word 78; word 170; word 104; word 198;\n word 78; word 166; word 40; word 198; word 78; word 93; word 133; word 117;\n word 78; word 192; word 132; word 109; word 78; word 195; word 132;\n word 109; word 110; word 173; word 195; word 87; word 79; word 30; word 192;\n word 87; word 79; word 76; word 133; word 117; word 110; word 104; word 128;\n word 107; word 79; word 173; word 37; word 21; word 79; word 196; word 39;\n word 21; word 79; word 124; word 208; word 123; word 79; word 189; word 65;\n word 71; word 111; word 128; word 64; word 71; word 111; word 138; word 217;\n word 91; word 79; word 155; word 137; word 75; word 79; word 136; word 67;\n word 71; word 111; word 2; word 132; word 125; word 110; word 13; word 132;\n word 125; word 78; word 91; word 65; word 71; word 111; word 87; word 128;\n word 75; word 79; word 80; word 208; word 91; word 79; word 132; word 4;\n word 0; word 209; word 26; word 133; word 123; word 110; word 17; word 133;\n word 123; word 78; word 23; word 66; word 71; word 111; word 72; word 131;\n word 75; word 79; word 86; word 211; word 91; word 79; word 51; word 194;\n word 87; word 79; word 119; word 8; word 128; word 61; word 116; word 16;\n word 192; word 61; word 106; word 38; word 21; word 79; word 101; word 20;\n word 192; word 61; word 81; word 65; word 71; word 111; word 119; word 24;\n word 192; word 61; word 107; word 28; word 192; word 61; word 137; word 106;\n word 133; word 78; word 155; word 42; word 133; word 78; word 230; word 106;\n word 139; word 78; word 239; word 42; word 139; word 78; word 90; word 16;\n word 192; word 61; word 112; word 107; word 207; word 78; word 100; \n word 43; word 207; word 78; word 35; word 105; word 198; word 78; word 53;\n word 41; word 198; word 78; word 27; word 134; word 99; word 110; word 134;\n word 132; word 117; word 110; word 93; word 8; word 192; word 61; word 83;\n word 12; word 192; word 61; word 76; word 20; word 192; word 61; word 113;\n word 4; word 128; word 61; word 211; word 180; word 115; word 110; word 98;\n word 183; word 108; word 110; word 105; word 159; word 122; word 78;\n word 218; word 156; word 125; word 78; word 18; word 134; word 99; word 78;\n word 157; word 132; word 117; word 78; word 73; word 64; word 71; word 111;\n word 122; word 66; word 71; word 111; word 64; word 4; word 192; word 61;\n word 75; word 4; word 198; word 60; word 184; word 135; word 114; word 110;\n word 94; word 135; word 105; word 110; word 170; word 135; word 114;\n word 78; word 2; word 183; word 96; word 110; word 18; word 159; word 107;\n word 78; word 208; word 159; word 107; word 78; word 217; word 183; \n word 96; word 110; word 76; word 135; word 105; word 78; word 82; word 64;\n word 71; word 111; word 43; word 4; word 193; word 60; word 48; word 67;\n word 71; word 111; word 90; word 105; word 140; word 78; word 200; word 66;\n word 71; word 111; word 69; word 41; word 140; word 78; word 88; word 106;\n word 144; word 78; word 89; word 42; word 144; word 78; word 104; word 12;\n word 128; word 61; word 68; word 107; word 216; word 78; word 94; word 43;\n word 216; word 78; word 180; word 104; word 217; word 78; word 163; \n word 40; word 217; word 78; word 133; word 134; word 100; word 78; word 96;\n word 132; word 126; word 78; word 143; word 134; word 100; word 110;\n word 161; word 192; word 87; word 79; word 24; word 192; word 87; word 79;\n word 114; word 132; word 126; word 110; word 241; word 217; word 91;\n word 79; word 48; word 36; word 21; word 79; word 29; word 39; word 21;\n word 79; word 72; word 130; word 107; word 79; word 5; word 66; word 71;\n word 111; word 160; word 67; word 71; word 111; word 68; word 210; word 123;\n word 79; word 251; word 137; word 75; word 79; word 109; word 4; word 132;\n word 60; word 29; word 132; word 101; word 110; word 136; word 64; word 71;\n word 111; word 59; word 66; word 71; word 111; word 176; word 211; word 91;\n word 79; word 183; word 131; word 75; word 79; word 13; word 132; word 101;\n word 78; word 132; word 4; word 0; word 209; word 228; word 245; word 255;\n word 181; word 18; word 133; word 123; word 110; word 0; word 133; word 123;\n word 78; word 23; word 66; word 71; word 111; word 91; word 130; word 75;\n word 79; word 8; word 192; word 87; word 79; word 90; word 210; word 91;\n word 79; word 109; word 4; word 132; word 60; word 16; word 37; word 21;\n word 79; word 91; word 67; word 71; word 111; word 119; word 0; word 158;\n word 60; word 0; word 66; word 71; word 111; word 123; word 0; word 159;\n word 60; word 96; word 0; word 157; word 60; word 132; word 0; word 128;\n word 210; word 32; word 4; word 194; word 60; word 33; word 0; word 223;\n word 60; word 5; word 64; word 128; word 82; word 189; word 12; word 2;\n word 78; word 5; word 118; word 130; word 82; word 190; word 12; word 2;\n word 78; word 31; word 48; word 192; word 61; word 21; word 32; word 192;\n word 61; word 12; word 64; word 192; word 61; word 173; word 134; word 127;\n word 110; word 9; word 96; word 192; word 61; word 187; word 129; word 65;\n word 79; word 3; word 80; word 192; word 61; word 10; word 112; word 192;\n word 61; word 143; word 133; word 99; word 78; word 156; word 133; word 99;\n word 110; word 38; word 133; word 106; word 78; word 5; word 16; word 192;\n word 61; word 227; word 133; word 102; word 110; word 42; word 133;\n word 106; word 110; word 191; word 134; word 127; word 78; word 108;\n word 216; word 80; word 79; word 115; word 136; word 64; word 79; word 88;\n word 137; word 65; word 79; word 3; word 0; word 192; word 61; word 74;\n word 217; word 81; word 79; word 181; word 209; word 81; word 79; word 147;\n word 65; word 71; word 111; word 109; word 132; word 101; word 110; \n word 88; word 65; word 71; word 111; word 187; word 66; word 71; word 111;\n word 181; word 217; word 112; word 79; word 108; word 132; word 101;\n word 78; word 169; word 137; word 96; word 79; word 234; word 133; word 102;\n word 78; word 141; word 133; word 127; word 78; word 20; word 68; word 192;\n word 61; word 169; word 66; word 71; word 111; word 165; word 133; word 106;\n word 78; word 25; word 100; word 192; word 61; word 182; word 180; word 126;\n word 110; word 143; word 131; word 97; word 79; word 163; word 133;\n word 106; word 110; word 138; word 211; word 113; word 79; word 38;\n word 133; word 123; word 110; word 141; word 133; word 127; word 110;\n word 112; word 136; word 97; word 79; word 79; word 65; word 71; word 111;\n word 170; word 209; word 112; word 79; word 191; word 129; word 96; \n word 79; word 41; word 133; word 123; word 78; word 236; word 133; word 120;\n word 78; word 13; word 84; word 192; word 61; word 95; word 65; word 71;\n word 111; word 234; word 133; word 120; word 110; word 142; word 134;\n word 109; word 78; word 156; word 134; word 109; word 110; word 237;\n word 135; word 115; word 110; word 244; word 135; word 115; word 78;\n word 123; word 216; word 113; word 79; word 72; word 137; word 64; word 79;\n word 74; word 217; word 80; word 79; word 56; word 133; word 108; word 78;\n word 23; word 116; word 192; word 61; word 31; word 20; word 192; word 61;\n word 43; word 135; word 119; word 78; word 209; word 128; word 96; word 79;\n word 72; word 65; word 71; word 111; word 21; word 52; word 192; word 61;\n word 132; word 8; word 0; word 209; word 195; word 133; word 107; word 110;\n word 10; word 36; word 192; word 61; word 98; word 216; word 80; word 79;\n word 114; word 136; word 64; word 79; word 35; word 135; word 119; word 110;\n word 90; word 133; word 117; word 110; word 87; word 133; word 117; \n word 78; word 112; word 67; word 71; word 111; word 100; word 136; word 65;\n word 79; word 83; word 131; word 65; word 79; word 79; word 211; word 81;\n word 79; word 27; word 4; word 192; word 61; word 117; word 216; word 81;\n word 79; word 82; word 64; word 71; word 111; word 162; word 156; word 125;\n word 78; word 195; word 208; word 112; word 79; word 106; word 135;\n word 127; word 110; word 164; word 66; word 71; word 111; word 243; \n word 65; word 71; word 111; word 70; word 217; word 112; word 79; word 90;\n word 137; word 96; word 79; word 106; word 135; word 127; word 78; word 223;\n word 133; word 107; word 78; word 16; word 64; word 128; word 61; word 113;\n word 64; word 71; word 111; word 80; word 133; word 119; word 78; word 3;\n word 72; word 192; word 61; word 43; word 134; word 104; word 110; word 194;\n word 66; word 71; word 111; word 218; word 64; word 71; word 111; word 118;\n word 209; word 80; word 79; word 5; word 134; word 127; word 78; word 107;\n word 129; word 64; word 79; word 25; word 104; word 192; word 61; word 21;\n word 56; word 192; word 61; word 203; word 66; word 71; word 111; word 175;\n word 209; word 80; word 79; word 182; word 180; word 126; word 110;\n word 142; word 131; word 97; word 79; word 27; word 134; word 127; word 110;\n word 70; word 135; word 115; word 110; word 63; word 134; word 104; \n word 78; word 136; word 211; word 113; word 79; word 28; word 183; word 126;\n word 110; word 112; word 139; word 97; word 79; word 31; word 48; word 128;\n word 61; word 81; word 133; word 119; word 110; word 11; word 112; word 128;\n word 61; word 14; word 65; word 71; word 111; word 2; word 4; word 129;\n word 60; word 11; word 159; word 125; word 78; word 56; word 210; word 112;\n word 79; word 49; word 130; word 96; word 79; word 183; word 129; word 64;\n word 79; word 45; word 133; word 108; word 110; word 73; word 135; word 115;\n word 78; word 204; word 133; word 100; word 78; word 186; word 217;\n word 113; word 79; word 168; word 137; word 97; word 79; word 13; word 84;\n word 192; word 61; word 139; word 67; word 71; word 111; word 72; word 67;\n word 71; word 111; word 218; word 133; word 100; word 110; word 17; \n word 67; word 71; word 111; word 110; word 132; word 109; word 78; word 8;\n word 76; word 128; word 61; word 124; word 132; word 109; word 110; \n word 11; word 12; word 128; word 61; word 45; word 134; word 114; word 110;\n word 20; word 28; word 128; word 61; word 52; word 134; word 114; word 78;\n word 123; word 219; word 113; word 79; word 247; word 65; word 71; word 111;\n word 72; word 139; word 64; word 79; word 82; word 219; word 80; word 79;\n word 56; word 133; word 108; word 78; word 23; word 92; word 128; word 61;\n word 23; word 116; word 192; word 61; word 31; word 20; word 192; word 61;\n word 43; word 135; word 119; word 78; word 209; word 128; word 96; word 79;\n word 72; word 66; word 71; word 111; word 132; word 4; word 0; word 209;\n word 164; word 245; word 255; word 181; word 163; word 129; word 64;\n word 79; word 173; word 209; word 80; word 79; word 112; word 67; word 71;\n word 111; word 207; word 133; word 107; word 78; word 219; word 133;\n word 107; word 110; word 26; word 183; word 126; word 110; word 2; word 159;\n word 125; word 78; word 171; word 156; word 125; word 78; word 36; word 135;\n word 119; word 110; word 46; word 133; word 108; word 110; word 138;\n word 131; word 97; word 79; word 137; word 211; word 113; word 79; word 203;\n word 66; word 71; word 111; word 220; word 208; word 112; word 79; word 20;\n word 32; word 128; word 61; word 6; word 36; word 192; word 61; word 145;\n word 67; word 71; word 111; word 163; word 65; word 71; word 111; word 217;\n word 132; word 117; word 110; word 212; word 132; word 117; word 78;\n word 44; word 134; word 104; word 110; word 53; word 134; word 104; \n word 78; word 18; word 4; word 192; word 61; word 16; word 64; word 128;\n word 61; word 118; word 139; word 64; word 79; word 88; word 134; word 127;\n word 78; word 92; word 134; word 127; word 110; word 104; word 219; \n word 80; word 79; word 23; word 135; word 116; word 78; word 66; word 67;\n word 71; word 111; word 134; word 136; word 65; word 79; word 241; word 134;\n word 111; word 110; word 250; word 134; word 111; word 78; word 143;\n word 216; word 81; word 79; word 223; word 217; word 113; word 79; word 206;\n word 137; word 97; word 79; word 42; word 65; word 71; word 111; word 11;\n word 4; word 129; word 60; word 55; word 211; word 81; word 79; word 3;\n word 92; word 128; word 61; word 50; word 131; word 65; word 79; word 25;\n word 135; word 116; word 110; word 69; word 159; word 125; word 78;\n word 152; word 139; word 96; word 79; word 21; word 44; word 128; word 61;\n word 242; word 66; word 71; word 111; word 148; word 219; word 112; \n word 79; word 230; word 65; word 71; word 111; word 79; word 183; word 126;\n word 110; word 59; word 211; word 112; word 79; word 22; word 65; word 71;\n word 111; word 152; word 66; word 71; word 111; word 73; word 133; word 102;\n word 110; word 52; word 131; word 96; word 79; word 75; word 133; word 102;\n word 78; word 21; word 135; word 114; word 110; word 60; word 217; word 80;\n word 79; word 51; word 137; word 64; word 79; word 168; word 130; word 96;\n word 79; word 163; word 210; word 112; word 79; word 2; word 12; word 128;\n word 61; word 9; word 135; word 114; word 78; word 116; word 67; word 71;\n word 111; word 147; word 67; word 71; word 111; word 60; word 138; word 97;\n word 79; word 59; word 218; word 113; word 79; word 229; word 65; word 71;\n word 111; word 104; word 64; word 71; word 111; word 138; word 134;\n word 118; word 110; word 34; word 133; word 107; word 78; word 238; \n word 67; word 71; word 111; word 25; word 133; word 115; word 110; word 45;\n word 133; word 107; word 110; word 67; word 209; word 80; word 79; word 79;\n word 129; word 64; word 79; word 58; word 131; word 64; word 79; word 57;\n word 211; word 80; word 79; word 75; word 156; word 125; word 78; word 144;\n word 134; word 118; word 78; word 5; word 4; word 129; word 60; word 58;\n word 67; word 71; word 111; word 87; word 180; word 126; word 110; word 111;\n word 64; word 71; word 111; word 177; word 137; word 97; word 79; word 26;\n word 108; word 128; word 61; word 235; word 66; word 71; word 111; word 21;\n word 133; word 115; word 78; word 16; word 28; word 128; word 61; word 166;\n word 217; word 113; word 79; word 11; word 12; word 128; word 61; word 132;\n word 209; word 80; word 79; word 15; word 92; word 128; word 61; word 209;\n word 64; word 71; word 111; word 14; word 72; word 128; word 61; word 152;\n word 129; word 64; word 79; word 21; word 44; word 128; word 61; word 124;\n word 67; word 71; word 111; word 17; word 76; word 128; word 61; word 152;\n word 64; word 71; word 111; word 28; word 60; word 128; word 61; word 24;\n word 104; word 128; word 61; word 232; word 39; word 64; word 109; word 234;\n word 47; word 65; word 109; word 236; word 55; word 66; word 109; word 238;\n word 63; word 67; word 109; word 255; word 3; word 1; word 145; word 192;\n word 3; word 95; word 214]")
mlkem/mlkem_intt.native had errors!
make: *** [Makefile:124: mlkem/mlkem_intt.correct] Error 1
ERROR > HOL_LIGHT (1/1)    None        (native):            FAILED (after 183s)
1 tests FAILED
```

hanno-becker commented:
Sorry @mkannwischer, I must have misclicked -- obviously this PR is not ready.

hanno-becker committed:
CI: Interleave AArch64 and x86_64 CT tests

This commit swaps the order of parameters in the job strategy
for the constant time CI tests, so that not all x86_64 tests
are run before all AArch64 tests. Instead, they are run in pairs.

Signed-off-by: Hanno Becker <beckphan@amazon.co.uk>

hanno-becker commented:
This doesn't work -- if we don't include bounds pre-conditions, the upfront normalization is needed to reduce the coefficients.
